yapapi/runner/__init__.py

```diff
@@ -234,23 +234,63 @@ async def accept_payment_for_agreement(agreement_id: str, *, partial=False) -> b
             await inv.accept(amount=inv.amount, allocation=allocation)
             return True
 
+        res_type_to_readable = {
+            'sub': "Subscription",
+            'prop': "Proposal",
+            'agr': "Agreement",
+            'act': "Activity",
+            'wkr': "Worker",
+            'task': "Task",
+        }
+
+        ev_type_to_readable = {
+            'recv': 'received',
+            'fail-subscribe': "failed to subscribe to market events",
+            'fail-collect': "failed to collect proposal events",
+            'fail-create': "could not be created on provider",
+            'fail-run': "cannot run commands on provider",
+            'respond': "answered",
+            'get-work': "getting work",
+            'get-results': "getting batch results",
+            'create': "created",
+            'batch-done': "batch done",
+            'confirm': "confirmed and approved",
+            'fail': "failed",
+            'accept': "accepted",
+            'payment_prep': "related payment prepared",
+            'payment_queued': "related payment queued",
+            'payment_accept': "related payment accepted",
+        }
+
         async def _tmp_log():
             while True:
                 item = await event_queue.get()
-                print(item)
+                res_text = res_type_to_readable[item[0]] if item[0] in res_type_to_readable else item[0]
+                ev_text = ev_type_to_readable[item[1]] if item[1] in ev_type_to_readable else item[1]
+                print(f"{res_text} {ev_text}, id: {item[2]}, info={item[3]}")
 
         def emit_progress(
-            resource_type: Literal["sub", "prop", "agr", "act", "wkr"],
+            resource_type: Literal["sub", "prop", "agr", "act", "wkr", "task"],
```

Comment:
I reckon the resource type would be better represented by an enum instead. This would allow for more readable names and better typing and IDE support (enum class members versus raw strings).
Additionally, enum values can be strings, so we could remove the dict `res_type_to_readable` above by having the appropriate names as values. Example implementation (based on [enum docs](https://docs.python.org/3/library/enum.html#using-automatic-values)):
```
from enum import Enum, auto

class CapitalNameEnum(Enum):
    def _generate_next_value_(name, start, count, last_values):
        return str(name).capitalize()

class ResourceType(CapitalNameEnum):

    activity = auto()
    agreement = auto()
    ...
```

Enum members (e.g. `ResourceType.activity`) are hashable, so they can be used as dict keys. The base class `CapitalNameEnum` modifies the behaviour of `auto` so that it returns a capitalized version of the member's name, so that e.g. `ResourceType.activity.value` returns `"Activity"`.

Reply:
Since changes from https://github.com/golemfactory/yapapi/pull/20 already depend on this branch I think it would be best to apply the changes I suggested here separately (as they touch a large part of this codebase).

Reply:
I can also include add the changes in #20 then merge them into this branch

Reply:
> Since changes from #20 already depend on this branch I think it would be best to apply the changes I suggested here separately (as they touch a large part of this codebase).

There is also unmerged pull request #16 which IMHO has to be merged first.
I agree that string literals are not the best solution here.